src/algorithms/meta/SubDivideFunctors.h

```diff
@@ -66,19 +70,19 @@ class GeometrySplit {
 private:
 
     void init() const {
-        m_id_dec = algorithms::GeoSvc::instance().detector()->readout(m_readout).idSpec().decoder();
+        *m_id_dec = algorithms::GeoSvc::instance().detector()->readout(m_readout).idSpec().decoder();
         for (auto d : m_divisions){
-            m_div_ids.push_back(m_id_dec->index(d));
+            m_div_ids->push_back((*m_id_dec)->index(d));
         }
     }
 
     std::vector<std::vector<long int>> m_ids;
     std::vector<std::string> m_divisions;
     std::string m_readout;
 
-    mutable std::shared_ptr<std::once_flag> is_init = std::make_shared<std::once_flag>();
-    mutable dd4hep::DDSegmentation::BitFieldCoder* m_id_dec;
-    mutable std::vector<size_t> m_div_ids;
+    std::shared_ptr<std::once_flag> is_init;
+    std::shared_ptr<dd4hep::DDSegmentation::BitFieldCoder*> m_id_dec;
```

Comment:
We don't need to store this on the heap
```suggestion
    dd4hep::DDSegmentation::BitFieldCoder m_id_dec;
```

Reply:
Okay, maybe, I get why.